### PR TITLE
fix(curriculum): Improve verbiage and add examples to the map, filter, and reduce challenges

### LINKED ISSUE #37704: Javascript map, filter, and reduce methods need examples and explanation of using callback functions
**Describe your problem and how to reproduce it:**
In several places in the "Functional Programming" challenges, we're asked to work with the data inside an array passed as an argument, without modifying the array.

In some of these challenges, it asks us to use `map`, `filter`, and `reduce`.

I think it would help a lot to have a lesson (maybe a series of lessons) explaining in more detail how each of those methods work.

**Add a Link to the page with the problem:**

Just one example:
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: 78.0.3904.87 
* Operating System: Windows 10

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #37704

This PR attempts to explain in detail how each of the methods introduced in the following challenges work and includes examples of how to use them.  The current versions of these challenges almost assume the user has already been taught how to use them, which is why the forum gets a lot of questions for these challenges.

* [Use the map Method to Extract Data from an Array](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array)

* [Use the filter Method to Extract Data from an Array](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array)

* [Use the reduce Method to Analyze Data](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data)

I also formatted the challenge seed code to use two space indentation, to reduce the line width of the code and make it more readable.

### CODE REVIEW

**Comment by RandellDawson** [comment]
@raisedadead Why does Gitpod still show Pending status?  When I click on Details link, it opens up Gitpod and creates the build and eventually shows everything working as it should.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md [hunk 1]
```diff
@@ -11,7 +11,21 @@ So far we have learned to use pure functions to avoid side effects in a program.
...
-Remember that the <code>map</code> method is a way to iterate over each item in an array. It creates a new array (without changing the original one) after applying a callback function to every element.
+The <code>map</code> method iterates over each item in an array and returns a new array (without mutating the original) after applying a callback function to every element.
+The callback function accepts three arguments (with names of your choice).  While iterating over the array, the arguments can be referenceed to change value of the  elements in the array returned.  The first argument is the current element being processed.  The second argument is the index of the current element being processed.  The third argument is the actual array upon which the map method was called.
+See below for an example using the <code>map</code> method on a <code>users</code> array containing objects to return a new array containing only the names of the users as elements. The example only uses the first argument, but there are times when you may need to reference the second and/or third arguments.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]:
```suggestion
See below for an example using the <code>map</code> method on the <code>users</code> array to return a new array containing only the names of the users as elements. For simplicity the example only uses the first argument of the callback.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 2]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
 
 ## Description
 <section id='description'>
+
 <code>Array.prototype.reduce()</code>, or simply <code>reduce()</code>, is the most general of all array operations in JavaScript. You can solve almost any array processing problem using the <code>reduce</code> method.
-This is not the case with the <code>filter</code> and <code>map</code> methods since they do not allow interaction between two different elements of the array. For example, if you want to compare elements of the array, or add them together, <code>filter</code> or <code>map</code> could not process that.
+
 The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as a special application of <code>reduce</code>.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as special applications of <code>reduce</code>.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 3]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
-However, before we get there, let's practice using <code>reduce</code> first.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is acheived via a callback function that returns a value created by the user's logic after each iteration.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called each iteration.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 4]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
+
+The callback function accepts four arguments (with names of your choice).  While iterating over the array, the arguments can be referenced in the callback function.  The first argument is known as the accumulator, which gets assigned the return value of the callback function during each iteration.  The second argument is the current element being processed.  The third argument is the index of the current element being processed.  The fourth argument is the actual array upon which the <code>reduce</code> method was called.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
The callback function accepts four arguments.  The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration.  The second is the current element being processed.  The third is the index of that element.  The fourth is the array upon which <code>reduce</code> was called.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 5]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
+
+In addition to the callback function, the <code>reduce</code>method has an additional parameter which takes an initial value for the accumulator.  If this second parameter is not used, then the accumulator's initial value is the first element of the array and the second argument is the second element in the first iteration.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
In addition to the callback function, <code>reduce</code> has an additional parameter which takes an initial value for the accumulator.  If this second parameter is not used, then the first iteration is skipped and the second iteration gets passed the first element of the array as the accumulator.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 6]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
+
+See below for an example using the <code>reduce</code> method on a <code>users</code> array containing objects to return the sum of all the users' ages. The example only uses the first and second arguments, but there are times when you may need to reference the third and/or fourth arguments.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
See below for an example using <code>reduce</code> on the <code>users</code> array to return the sum of all the users' ages. For simplicity the example only uses the first and second arguments.
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
@RandellDawson since you wanted the verbiage shrinking I ended up being quite aggressive with the cuts/re-wording.  Let me know if I went too far.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md [hunk 7]
```diff
@@ -7,8 +7,22 @@ forumTopicId: 18179
...
+The <code>filter</code> method iterates over each element in an array and returns a new array of the same or lesser length after applying a callback function to every element.  The callback function tests each element (with the user's logic) to return either `true` to keep an element or `false` to remove the element.  The method can not alter the original value of the elements.
+The callback function accepts three arguments (with names of your choice).  While iterating over the array, the arguments can be referenceed to change the elements in the array returned.  The first argument is the current element being processed.  The second argument is the index of the current element being processed.  The third argument is the actual array upon which the <code>filter</code> method was called.
+See below for an example using the <code>filter</code> method on a <code>users</code> array containing objects to return a new array containing only the users under the age of 30 as elements. The example only uses the first argument, but there are times when you may need to reference the second and/or third arguments.
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]: Can you add a comma after `simplicity` so I can commit this?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md [hunk 8]
```diff
@@ -7,8 +7,22 @@ forumTopicId: 18179
 
 ## Description
 <section id='description'>
-Another useful array function is <code>Array.prototype.filter()</code>, or simply <code>filter()</code>. The <code>filter</code> method returns a new array which is at most as long as the original array, but usually has fewer items.
-<code>Filter</code> doesn't alter the original array, just like <code>map</code>. It takes a callback function that applies the logic inside the callback on each element of the array. If an element returns true based on the criteria in the callback function, then it is included in the new array.
+Another useful array function is <code>Array.prototype.filter()</code>, or simply <code>filter()</code>. 
+The <code>filter</code> method iterates over each element in an array and returns a new array of the same or lesser length after applying a callback function to every element.  The callback function tests each element (with the user's logic) to return either `true` to keep an element or `false` to remove the element.  The method can not alter the original value of the elements.
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]: Can you add a comma after `In other words` so I can commit this?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md [hunk 9]
```diff
@@ -11,7 +11,21 @@ So far we have learned to use pure functions to avoid side effects in a program.
...
 Let's start with some simple array functions, which are methods on the array object prototype. In this exercise we are looking at <code>Array.prototype.map()</code>, or more simply <code>map</code>.
-Remember that the <code>map</code> method is a way to iterate over each item in an array. It creates a new array (without changing the original one) after applying a callback function to every element.
+The <code>map</code> method iterates over each item in an array and returns a new array (without mutating the original) after applying a callback function to every element.
+The callback function accepts three arguments (with names of your choice).  While iterating over the array, the arguments can be referenceed to change value of the  elements in the array returned.  The first argument is the current element being processed.  The second argument is the index of the current element being processed.  The third argument is the actual array upon which the map method was called.
```
[the remark this answers is not in the document]
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]: Can you add a comma after `When the callback is used` so I can commit this?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 6 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: Can you add a comma after `For simplicity` so I can commit this?

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by raisedadead** [comment]
> Why does Gitpod still show Pending status? When I click on Details link, it opens up Gitpod and creates the build and eventually shows everything working as it should.

Maybe it's just a glitch or something. 🤷‍♂

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 10]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
-However, before we get there, let's practice using <code>reduce</code> first.
+
+The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as special applications of <code>reduce</code>.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called each iteration.
+
+The callback function accepts four arguments.  The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration.  The second is the current element being processed.  The third is the index of that element.  The fourth is the array upon which <code>reduce</code> was called.
+
+In addition to the callback function, <code>reduce</code> has an additional parameter which takes an initial value for the accumulator.  If this second parameter is not used, then the first iteration is skipped and the second iteration gets passed the first element of the array as the accumulator.
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: These lines have two spaces after each sentence. Please fix that.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 11]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
-The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as a special application of <code>reduce</code>.
-However, before we get there, let's practice using <code>reduce</code> first.
+
+The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as special applications of <code>reduce</code>.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called each iteration.
+
+The callback function accepts four arguments.  The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration.  The second is the current element being processed.  The third is the index of that element.  The fourth is the array upon which <code>reduce</code> was called.
```
Comment (thecodingaviator) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
The callback function accepts four arguments: The first argument is known as the accumulator, which gets assigned the return value of the callback function from the previous iteration, the second is the current element (the element being processed), the third is the index of that element and the fourth is the array upon which <code>reduce</code> is called.
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 10 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: Not sure what you mean.  See below for how it renders locally.

image

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 11 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: I am not sure about replace the period with a colon here.  Not sure why we can not just leave it as a period.  It is a simple sentence as it is.

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: Looks better now?

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: I honestly don't see a problem with what was originally there.

Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: I felt like the statement you had chosen to describe the second parameter is a bit confusing. See the change I’ve made to that part of the sentence.

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: I feel using `current element` and then in parentheses using the word `the element` is redundant.  I am OK with the commas and changing the `was` to `is` in the last part of the sentence though.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md [hunk 12]
```diff
@@ -11,7 +11,21 @@ So far we have learned to use pure functions to avoid side effects in a program.
 This is only the beginning. As its name suggests, functional programming is centered around a theory of functions.
 It would make sense to be able to pass them as arguments to other functions, and return a function from another function. Functions are considered <dfn>first class objects</dfn> in JavaScript, which means they can be used like any other object. They can be saved in variables, stored in an object, or passed as function arguments.
 Let's start with some simple array functions, which are methods on the array object prototype. In this exercise we are looking at <code>Array.prototype.map()</code>, or more simply <code>map</code>.
-Remember that the <code>map</code> method is a way to iterate over each item in an array. It creates a new array (without changing the original one) after applying a callback function to every element.
+The <code>map</code> method iterates over each item in an array and returns a new array containing the results of calling the callback function on each element.  It does this without mutating the original.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]:
```suggestion
The <code>map</code> method iterates over each item in an array and returns a new array containing the results of calling the callback function on each element.  It does this without mutating the original array.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md [hunk 13]
```diff
@@ -11,7 +11,21 @@ So far we have learned to use pure functions to avoid side effects in a program.
...
+  { name: 'camperCat', age: 10 }
+];
+
+const names = users.map(user => user.name);
+console.log(names); // [ 'John', 'Amy', 'camperCat' ]
+```
+
 </section>
 
 ## Instructions
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]:
```
Use <code>map</code> to pull the title and rating from <code>watchList</code> and save the new array in the <code>rating</code> variable.
```

This instruction should probably be changed. It sounds like we want users to pull the title and rating properties directly into a new array. While the task is actually to return a new "object" from the `.map()`, with `title` and `rating` properties, containing the corresponding values from each object. This array of new objects should be saved in the `watchList` variable.

Yes, there is the for-loop in the example, which you could probably look at to deduce this, but instructions should ideally be self-explanatory.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md [hunk 14]
```diff
@@ -7,8 +7,22 @@ forumTopicId: 18179
 
 ## Description
 <section id='description'>
-Another useful array function is <code>Array.prototype.filter()</code>, or simply <code>filter()</code>. The <code>filter</code> method returns a new array which is at most as long as the original array, but usually has fewer items.
-<code>Filter</code> doesn't alter the original array, just like <code>map</code>. It takes a callback function that applies the logic inside the callback on each element of the array. If an element returns true based on the criteria in the callback function, then it is included in the new array.
+Another useful array function is <code>Array.prototype.filter()</code>, or simply <code>filter()</code>. 
+<code>filter</code> calls a function on each element of an array and returns a new array containing only the elements for which that function returns <code>true</code>.  In other words, it filters the array, based on the function passed to it.  Like <code>map</code> it does this without needing to modify the original array.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]:
```suggestion
<code>filter</code> calls a function on each element of an array and returns a new array containing only the elements for which that function returns <code>true</code>.  In other words, it filters the array, based on the function passed to it.  Like <code>map</code>, it does this without needing to modify the original array.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md [hunk 15]
```diff
@@ -7,8 +7,22 @@ forumTopicId: 18179
...
+  { name: 'camperCat', age: 10 }
+];
+
+const usersUnder30 = users.filter(user => user.age < 30);
+console.log(usersUnder30); // [ { name: 'Amy', age: 20 }, { name: 'camperCat', age: 10 } ]
+```
+
 </section>
 
 ## Instructions
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]: Would be a good idea to add to the instructions that the resulting array needs to be saved in the `filteredList` variable.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 16]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
-This is not the case with the <code>filter</code> and <code>map</code> methods since they do not allow interaction between two different elements of the array. For example, if you want to compare elements of the array, or add them together, <code>filter</code> or <code>map</code> could not process that.
-The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as a special application of <code>reduce</code>.
-However, before we get there, let's practice using <code>reduce</code> first.
+
+The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as special applications of <code>reduce</code>.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called each iteration.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is achieved via a callback function that is called on each iteration.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md [hunk 17]
```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
...
+See below for an example using <code>reduce</code> on the <code>users</code> array to return the sum of all the users' ages. For simplicity, the example only uses the first and second arguments.
+
+```js
+const users = [
+  { name: 'John', age: 34 },
+  { name: 'Amy', age: 20 },
+  { name: 'camperCat', age: 10 }
+];
+
+const u = users.reduce((sum, user) => sum + user.age, 0);
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]:
```suggestion
const sumOfAges = users.reduce((sum, user) => sum + user.age, 0);
```

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Changes for fixing a few bugs and some improvements.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 16 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: @Manish-Giri Should it be `on each` or `in each`?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 11 above)
Comment (thecodingaviator) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: Edited.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md (hunk 16 above)
Comment (Manish-Giri) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md]: @RandellDawson Either works, up to you. `during each` is also an option.

**Review by moT01 — APPROVED** [review]
I'm fine with these changes. I think it's a big improvement. Looks good @RandellDawson 🎉

**Comment by RandellDawson** [comment]
@moT01 I am still working on add @Manish-Giri's suggested changes.  I will try to do that today and then request one last look from you all.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md (hunk 13 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md]: @Manish-Giri I made a new commit that I think improves the instructions for this challenge.  Please let me know what you think.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md (hunk 15 above)
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]: @Manish-Giri I made a new commit that I think improves the instructions for this challenge.  Please let me know what you think.

**Comment by RandellDawson** [comment]
@Manish-Giri and @thecodingaviator Can you please take one last review to see if this is ready to merge?  Thanks.

**Review by Manish-Giri — APPROVED** [review]
@RandellDawson Looks good. Good job 💯

**Review by thecodingaviator — APPROVED** [review]

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**